src/main/java/com/couchbase/lite/support/Batcher.java

```diff
@@ -146,7 +146,9 @@ public void flushAll() {
     public void clear() {
         unschedule();
         inbox.clear();
-        inbox.notify();
+        synchronized (inbox) {
```

Comment:
Do we need to move `inbox.clear()` to inside synchronized block as multiple threads could access inbox?